packages/cursorless-engine/src/languages/TreeSitterQuery/queryPredicateOperators.ts

```diff
@@ -185,6 +185,23 @@ class AllowMultiple extends QueryPredicateOperator<AllowMultiple> {
   }
 }
 
+/** Indicates that it's okay for this scope to extend contiguously it through its siblings. */
+class Contiguous extends QueryPredicateOperator<Contiguous> {
+  name = "contiguous!" as const;
+  schema = z.union([z.tuple([q.node]), z.tuple([q.node, q.string])]);
+
+  run(nodeInfo: MutableQueryCapture, pattern?: string) {
```

Comment:
Yikes. This is a slippery slope. This operator merges to completely orthogonal concepts. I fear we're sleepwalking into the same problem we had with our old typescript-based matchers, where we had 5 custom functions per language.

Maybe we could add a predicate that rejects a pattern if it doesn't match a regex, and then define the pattern twice: once with positive match, one negative?

Reply:
This is still not language specific. Probably more languages that dont separate block comments in tree sitter.

That could also work, but is that really easier to read and maintain?

Reply:
Because the pattern matches a single line comment? look how I use it in js.

Lets.

Reply:
https://github.com/cursorless-dev/cursorless/labels/to%20discuss

Reply:
Update from meet-up:

- Remove `pattern` arg from this predicate
- use the built-in tree-sitter [`match` predicate](https://github.com/tree-sitter/tree-sitter/blob/master/lib/binding_web/binding.js#L906), and duplicate the pattern, once with `contiguous` and once without

Reply:
done!